docs/running-on-kubernetes.md

```diff
@@ -477,6 +477,22 @@ from the other deployment modes. See the [configuration page](configuration.html
   </td>
 </tr>
 <tr>
+  <td><code>spark.kubernetes.executor.labels</code></td>
+  <td>(none)</td>
+  <td>
+    Custom labels that will be added to the executor pods. This should be a comma-separated list of label key-value pairs,
+    where each label is in the format <code>key=value</code>.
```

Comment:
One possibility is to support any conf of the following format:

`spark.kubernetes.driver.label.myLabel`

where the value of this configuration is the value of the label.

Then in SparkConf we look for all configurations with the prefix `spark.kubernetes.driver.label`  and construct the labels map accordingly. I think in such a scheme we can support any string without escaping.